base/sequence_clock.go

```diff
@@ -291,6 +292,23 @@ func (c *SequenceClockImpl) UpdateWithClock(updateClock SequenceClock) {
 	}
 }
 
+func (c *SequenceClockImpl) UpdateWithPartitionClocks(partitionClocks []*PartitionClock, allowRollback bool) {
```

Comment:
SyncSequenceClock should be used by any consumers requiring a goroutine safe clock - SequenceClockImpl doesn't lock (already mentioned in the struct documentation for SequenceClockImpl).